pkg/geo/geo.go

```diff
@@ -133,7 +133,27 @@ func MakeGeometryUnsafe(spatialObject geopb.SpatialObject) Geometry {
 
 // MakeGeometryFromPointCoords makes a point from x, y coordinates.
 func MakeGeometryFromPointCoords(x, y float64) (Geometry, error) {
-	s, err := spatialObjectFromGeomT(geom.NewPointFlat(geom.XY, []float64{x, y}), geopb.SpatialObjectType_GeometryType)
+	return MakeGeometryFromLayoutAndPointCoords(geom.XY, []float64{x, y})
+}
+
+// MakeGeometryFromLayoutAndPointCoords makes a point with a given layout and ordered slice of coordinates.
+func MakeGeometryFromLayoutAndPointCoords(
+	layout geom.Layout, flatCoords []float64,
+) (Geometry, error) {
+	// Validate that the stride matches what is expected for the layout.
+	switch {
+	case layout == geom.XY && len(flatCoords) == 2:
+		break
+	case layout == geom.XYM && len(flatCoords) == 3:
+		break
+	case layout == geom.XYZ && len(flatCoords) == 3:
+		break
+	case layout == geom.XYZM && len(flatCoords) == 4:
+		break
```

Comment:
nit: you can omit all these breaks.

Reply:
Hmm is it ok if I leave these in? When I first saw an empty switch case in Go code, I thought it would fallthrough the same way they would in C so I like to have explicit `break` statements for clarity.

Reply:
not very go-like of you :P i'd prefer them out.